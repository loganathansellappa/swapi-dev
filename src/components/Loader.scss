
.inline {
  animation: circle 1.2s steps(8) infinite;
  color: azure;
  margin-left: 50%;
  margin-top: 10%;
}

@keyframes circle {
  from {
    transform: rotate(90deg);
  }

  to {
    transform: rotate(450deg);
  }
}

