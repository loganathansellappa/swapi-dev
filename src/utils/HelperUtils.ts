export const apiService = () => `${process.env.SWAP_URL}`;
export const capitalize = (s: string = 'n/a') => s[0].toUpperCase() + s.slice(1)