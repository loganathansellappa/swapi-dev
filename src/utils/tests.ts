import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';

export const renderWithRouter = (ui: JSX.Element, { route = '/' } = {}) => {
  window.history.pushState({}, 'Test page', route);

  return {
    user: userEvent.setup(),
    ...render(ui, { wrapper: BrowserRouter }),
  };
};
export const testStore = () => {
  const mockStore = configureStore();
  const initialState = { favoritePeople: { favourites: [] } };
  const store = mockStore(initialState);
  return store;
};

export const mockResource = () => {
  return {
    name: 'Luke Skywalker',
    height: '172',
    mass: '77',
    hair_color: 'blond',
    skin_color: 'fair',
    eye_color: 'blue',
    birth_year: '19BBY',
    gender: 'male',
    homeworld: 'https://swapi.dev/api/planets/1/',
    films: [
      'https://swapi.dev/api/films/1/',
      'https://swapi.dev/api/films/2/',
      'https://swapi.dev/api/films/3/',
      'https://swapi.dev/api/films/6/',
    ],
    species: [],
    vehicles: [
      'https://swapi.dev/api/vehicles/14/',
      'https://swapi.dev/api/vehicles/30/',
    ],
    starships: [
      'https://swapi.dev/api/starships/12/',
      'https://swapi.dev/api/starships/22/',
    ],
    created: '2014-12-09T13:50:51.644000Z',
    edited: '2014-12-20T21:17:56.891000Z',
    url: 'https://swapi.dev/api/people/1/',
  };
};
