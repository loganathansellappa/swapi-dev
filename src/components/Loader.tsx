import React from "react";
import {ImSpinner2} from "react-icons/all";
import "./Loader.scss";

interface LoaderProps {
    size?: number
    color?: string
    className?: string
}

export const Loader = ({size = 100, color = 'white', className = 'inline'}: LoaderProps) =>
    <div  >
        <ImSpinner2 size={size}  color={color} className={className}/>
    </div>
