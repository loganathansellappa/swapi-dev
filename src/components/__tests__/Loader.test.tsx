import Loader from '../Error';
import { renderWithRouter } from '../..//utils/tests';

test('renders Loader Page', () => {
    const { container } = renderWithRouter(<Loader />);
    expect(container.firstChild).toMatchSnapshot();
});
