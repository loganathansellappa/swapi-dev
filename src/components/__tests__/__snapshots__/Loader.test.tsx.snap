// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Loader Page 1`] = `
<div>
  <p>
    Something Went wrong 
  </p>
  <p>
    400
  </p>
  <button>
    Go Home
  </button>
</div>
`;
